pkg/descheduler/descheduler.go

```diff
@@ -283,18 +284,25 @@ func RunDeschedulerStrategies(ctx context.Context, rs *options.DeschedulerServer
 			deschedulerPolicy.MaxNoOfPodsToEvictPerNode,
 			deschedulerPolicy.MaxNoOfPodsToEvictPerNamespace,
 			nodes,
-			getPodsAssignedToNode,
-			evictLocalStoragePods,
-			evictSystemCriticalPods,
-			ignorePvcPods,
-			evictBarePods,
 			!rs.DisableMetrics,
 		)
 
 		for name, strategy := range deschedulerPolicy.Strategies {
 			if f, ok := strategyFuncs[name]; ok {
 				if strategy.Enabled {
-					f(ctx, rs.Client, strategy, nodes, podEvictor, getPodsAssignedToNode)
+					nodeFit := false
+					if name != "PodLifeTime" {
+						if strategy.Params != nil {
+							nodeFit = strategy.Params.NodeFit
+						}
+					}
```

Comment:
Ideally, in the future we should have the strategy initialized where the interface provides `NodeFit()` similar to [this](https://github.com/kubernetes-sigs/descheduler/pull/706/files#diff-67816189b07b98460b57a10b41280403c53a4dfda42fdd8f54475812cb99f56eR83)

Reply:
This is only a temporary hack to transition into the new v1alpha2 configuration where we can implement a builder/factory and run a validation to make sure the nodeFit gets set only where it makes sense.

Reply:
could this check still be in the strategy code?

Reply:
ah, is the answer no since `nodeFit` is being baked into the `evictorFilter` now? This will eventually be the type of thing that's in the strategy code though, right?

Reply:
Yeah. The `nodeFit` option/flag is going to be transform into the `DefaultEvictor` plugin argument. Anyone can then implement their own evictor `Filter` extension point. Making the descheduler unaware of the `nodeFit` option.